runtime/hsa-runtime/core/driver/kfd/amd_kfd_driver.cpp

```diff
@@ -42,40 +42,120 @@
 
 #include "core/inc/amd_kfd_driver.h"
 
-#include <sys/ioctl.h>
-
 #include <memory>
 #include <string>
 
+#include <link.h>
+#include <sys/ioctl.h>
+
 #include "hsakmt/hsakmt.h"
 
-#include "core/inc/amd_cpu_agent.h"
-#include "core/inc/amd_gpu_agent.h"
 #include "core/inc/amd_memory_region.h"
-#include "core/inc/exceptions.h"
 #include "core/inc/runtime.h"
 
+extern r_debug _amdgpu_r_debug;
+
 namespace rocr {
 namespace AMD {
 
 KfdDriver::KfdDriver(std::string devnode_name)
     : core::Driver(core::DriverType::KFD, devnode_name) {}
 
-hsa_status_t KfdDriver::Init() { return HSA_STATUS_SUCCESS; }
+hsa_status_t KfdDriver::Init() {
+  HSAKMT_STATUS ret = hsaKmtRuntimeEnable(
+      &_amdgpu_r_debug, core::Runtime::runtime_singleton_->flag().debug());
+
+  if (ret != HSAKMT_STATUS_SUCCESS && ret != HSAKMT_STATUS_NOT_SUPPORTED)
+    return HSA_STATUS_ERROR;
+
+  uint32_t caps_mask = 0;
+  hsaKmtGetRuntimeCapabilities(&caps_mask);
```

Comment:
Would this ever throw an error in the future (right now it returns always HSAKMT_STATUS_SUCCESS)?

Reply:
Fixed